File: backend-integration-test/src/test/java/hu/bhr/backend/customer/dto/CustomerResponse.java
package hu.bhr.backend.customer.dto;

import io.cucumber.core.internal.com.fasterxml.jackson.annotation.JsonProperty;

public record CustomerResponse(
    @JsonProperty("id") String id,
    @JsonProperty("firstName") String firstName,
    @JsonProperty("lastName") String lastName,
    @JsonProperty("nickname") String nickname,
    @JsonProperty("email") String email,
    @JsonProperty("phoneNumber") String phoneNumber,
    @JsonProperty("relationship") String relationship,
    @JsonProperty("createdAt") String createdAt,
    @JsonProperty("updatedAt") String updatedAt
) {}

File: crm-frontend/src/lib/models/customer-response.ts
export interface CustomerResponse {
  id: string;
  firstName: string;
  lastName: string;
  nickname: string;
  email: string;
  phoneNumber: string;
  relationship: string;
  residence: ResidenceResponse;
  createdAt: string;
  updatedAt: string;
}

export interface ResidenceResponse {
  residenceId: string;
  customerId: string;
  zipCode: string;
  streetAddress: string;
  addressLine2: string;
  city: string;
  country: string;
  createdAt: string;
  updatedAt: string;
}
